Update `current`

aleokdev opened the pull request:
Remember to do merge from Git

JtotheThree committed:
World File Parsing (#320)

* add world file parsing

* add world file parsing

* clarity and cleanup

* read world from reader

* add option to preload maps in world

* fix documentation, revert auto load maps

* remove tmx map variable from worldmap

* Remove automatic dir pattern testing

* formatting and better docs

* formatting and better docs

* Moved pattern utils to World impl

* Missed dep for world feature and cargo build step

* match_path and match_paths

* add match_path to WorldPattern

* reduce utf-8 checks on path iteration

* match_path_impl and readme update

* empty vecs instead of option

* empty vecs instead of option

* Oddities with readme

* fix load_world docs

* fix source not populating

bjorn reviewed (dismissed):
Thanks for making the 0.14 release!

bjorn commented:
What's up with this, btw?

```
 ---- src/lib.rs - (line 42) stdout ----
error: couldn't read `src/assets/tiled_csv.tmx`: No such file or directory (os error 2)
  --> src/lib.rs:56:37
   |
17 |             Ok(std::io::Cursor::new(include_bytes!("assets/tiled_csv.tmx")))
   |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `include_bytes` (in Nightly builds, run with -Z macro-backtrace for more info)
help: there is a file with the same name in a different directory
   |
17 |             Ok(std::io::Cursor::new(include_bytes!("../assets/tiled_csv.tmx")))
   |                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~

error: aborting due to 1 previous error

Couldn't compile the test.
```

aleokdev commented:
No clue. Maybe a new Rust version changed the `include` macro to be relative to the source file instead of the project folder. I'll try to fix it this afternoon

aleokdev commented:
Seems it's this: https://doc.rust-lang.org/edition-guide/rust-2024/rustdoc-nested-includes.html
While we're using edition 2018 currently, I'm not sure why the error is only being thrown now.

aleokdev commented:
Weird. The logs for build (push) which doesn't fail treats the doctests as being in the README:
![Screenshot_20250224_132910](https://github.com/user-attachments/assets/bc320a52-77e9-48e7-ada0-c40d80cf6d93)
Meanwhile build (pull_request) fails, and shows the tests as being in src/lib.rs:
![Screenshot_20250224_133009](https://github.com/user-attachments/assets/796327a9-a64f-430a-bee7-f63314efea74)

I'm not exactly sure what the difference between both builds is. Any ideas @bjorn?

adrien-bon commented:
Sorry for bothering you again, but do you have an ETA for the actual release ?
Is there something I can do to help ?
Thanks!

aleokdev commented:
Sorry, which release? If you are referring to the one that includes #320, I uploaded that one 3 weeks ago to crates.io: https://github.com/mapeditor/rs-tiled/releases/tag/v0.14.0 https://docs.rs/tiled/0.14.0/tiled/

This PR is just to update the current branch, but there are a few errors with the CI that we don't know how to fix. It isn't urgent so I haven't looked further into it.

adrien-bon commented:
Oh my bad, I saw this PR and though #320 was not yet published.
Sorry for the noise

aleokdev committed:
Fix CI (#326)

CI was failing with an error related to include! that it was not doing before.
This looks like a regression on someone else's part to be honest.
I'm not sure what we can do apart from changing the edition or just
skipping these tests, which is what I've done here.
